data/category.go

```diff
@@ -116,17 +119,24 @@ func (c *Category) Seed(cfg write.ClientConfig) (int, error) {
 	}
 }
 
-func (c *Category) Test(cfg write.ClientConfig) error {
+func (c *Category) Test(cfg write.ClientConfig) (err error) {
 	for _, s := range c.specs {
-		err := s.Test(cfg)
+		err = s.Test(cfg)
 		if err != nil {
-			fmt.Println(err)
+			fmt.Fprint(os.Stderr, err)
```

Comment:
I think rather than writing directly to stderr, a `Category` should have an associated SuccessRecorder and FailureRecorder. `(spec).Test` should not be aware of result recorders, but should just return its result.

Something like

```go
res, err := s.Test(cfg)
if err != nil {
  fmt.Fprintf(os.Stderr, "Error executing test: %v\n", err)
}
if res.Success {
 c.RecordSuccess(res.Name)
} else {
 c.RecordFailure(res.Failure)
}
```

Maybe not that API precisely, but something along those lines.  That will make it easier going forward to adjust output accordingly, and could potentially support jUnit XML, TAP, or whatever format other tooling needs to consume.